handler-proxy/src/main/java/io/netty/handler/proxy/ProxyHandler.java

```diff
@@ -287,7 +287,7 @@ private void setConnectSuccess() {
             connectTimeoutFuture.cancel(false);
         }
 
-        if (connectPromise.trySuccess(ctx.channel())) {
+        if (!connectPromise.isDone()) {
```

Comment:
If we expect this to be called frequently consider using a non-volatile variable to avoid duplicate processing. For example can we use `finished` instead?